pkgs/tools/archivers/7zz/default.nix

```diff
@@ -69,17 +77,13 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
-  doInstallCheck = true;
-
-  installCheckPhase = ''
-    runHook preInstallCheck
-
-    $out/bin/7zz --help | grep ${version}
-
-    runHook postInstallCheck
-  '';
-
-  passthru.updateScript = ./update.sh;
+  passthru = {
+    updateScript = ./update.sh;
+    tests.version = testVersion {
+      package = _7zz;
+      command = "7zz --help";
+    };
+  };
```

Comment:
> installCheckPhase is similar to checkPhase, in what both means that if this test fail I don't even want this package to be build

> But if the idea is test only the current package, installCheckPhase generally makes more sense to me (of course, there could be exceptions).

https://github.com/NixOS/rfcs/pull/119#discussion_r783597367

I'd like to keep the `installCheckPhase` as-is for now until RFC0119 is ironed out.

Reply:
Well, I followed most discussions in RFC0119 and it seems that the general perception is that `passthru.tests` and `installCheckPhase` should be considered the same: the package shouldn't be merged if it fails. @ofborg will automatically run `passthru.tests`, and nobody should merge packages that don't pass the tests without a good reason.

As an argument for `testVersion`: it does more than the previous test and it is more standard, so I am in favor of keeping it.

Reply:
For things `testVersion` does better than the previous test:

- It checks if the return code of `command` is 0, otherwise the test fails
- It uses the `grep -Fw` instead of just `grep`. This avoids matching something like `Random string with the version21.07mixed inside just on coincidence`.

Reply:
And for development, I found it sufficient to run `nix-build -A _7zz.tests` instead of `nix-build -A _7zz`. This will run the tests, but since the tests depend on the package itself, they will also retrigger the package rebuild.